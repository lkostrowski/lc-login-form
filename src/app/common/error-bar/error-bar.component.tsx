import cx from 'classnames';
import React, { HTMLProps } from 'react';

import { ReactComponent as ErrorIcon } from '../../../icons/error.svg';

import styles from './error-bar.module.scss';

interface Props extends HTMLProps<HTMLDivElement> {}

export const ErrorBar: React.FC<Props> = ({
    className,
    children,
    ...props
}) => (
    <div className={cx(className, styles.container)} {...props}>
        <ErrorIcon className={styles.icon} />
        {children}
    </div>
);
