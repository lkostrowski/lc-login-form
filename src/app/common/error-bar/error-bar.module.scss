@import '../../../styles/colors';
@import '../../../styles/vars';

.container {
    background: $lightRed;
    color: $white;
    font-size: 1.4rem;
    font-weight: 500;
    padding: 1rem 5rem;
    text-align: center;
    position: relative;
    border-radius: $borderRadius;
}

.icon {
    position: absolute;
    top: 50%;
    left: 1rem;
    transform: translateY(-50%);
}
