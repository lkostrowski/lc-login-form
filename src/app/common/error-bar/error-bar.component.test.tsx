import { cleanup, render } from '@testing-library/react';
import React from 'react';

import { ErrorBar } from './error-bar.component';

afterEach(cleanup);

describe('Error Bar component', () => {
    describe('Snapshots', () => {
        test('Default', () => {
            const { container } = render(<ErrorBar>Error content</ErrorBar>);

            expect(container).toMatchSnapshot();
        });
    });
});
