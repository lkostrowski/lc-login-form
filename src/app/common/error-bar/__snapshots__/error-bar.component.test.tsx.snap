// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Bar component Snapshots Default 1`] = `
<div>
  <div
    class="container"
  >
    <svg
      class="icon"
    >
      error.svg
    </svg>
    Error content
  </div>
</div>
`;
